js/src/wrappers/claude-agent-sdk.ts

```diff
@@ -0,0 +1,505 @@
+              ) as AsyncGenerator<SDKMessage, void, unknown>;
+
+            for await (const message of generator) {
+              const currentTime = getCurrentUnixTimestamp();
+
+              // Check if this is a new message ID
+              const messageId = message.message?.id;
+              if (messageId && messageId !== currentMessageId) {
+                // Create span for previous message group
+                await createLLMSpan();
+
+                // Start new message group
+                currentMessageId = messageId;
+                currentMessageStartTime = currentTime;
+              }
+
+              // Accumulate messages for the current message ID
+              if (message.type === "assistant" && message.message?.usage) {
+                currentMessages.push(message);
+              }
+
+              yield message;
+            }
+
+            // Create span for final message group
+            await createLLMSpan();
+
+            // Log final output to top-level span - just the last message content
+            span.log({
+              output:
+                finalResults.length > 0
+                  ? finalResults[finalResults.length - 1]
+                  : undefined,
+            });
+          } catch (error) {
+            span.log({
+              error: error instanceof Error ? error.message : String(error),
+            });
+            throw error;
+          } finally {
+            endSpan();
+          }
+        })();
+
+      // Return the generator directly - avoid Proxy for compatibility
+      return wrappedGenerator as ReturnType<T>;
+    },
+  });
+
+  return proxy;
+}
+
+/**
+ * Wraps a Claude Agent SDK tool definition to add Braintrust tracing for tool executions.
+ *
+ * @param toolDef - The tool definition created with the SDK's `tool()` function
+ * @returns A wrapped tool definition with traced handler
+ *
+ * @example
+ * ```typescript
+ * import { tool } from "@anthropic-ai/claude-agent-sdk";
+ * import { wrapClaudeAgentTool } from "braintrust";
+ * import { z } from "zod";
+ *
+ * const myTool = tool(
+ *   "calculator",
+ *   "Performs basic arithmetic",
+ *   { operation: z.enum(["add", "subtract"]), a: z.number(), b: z.number() },
+ *   async (args) => ({
+ *     content: [{ type: "text", text: String(args.a + args.b) }]
+ *   })
+ * );
+ *
+ * const wrappedTool = wrapClaudeAgentTool(myTool);
+ * ```
+ */
+export function wrapClaudeAgentTool<T>(
+  toolDef: SdkMcpToolDefinition<T>,
+): SdkMcpToolDefinition<T> {
+  const originalHandler = toolDef.handler;
+
+  const wrappedHandler: ToolHandler<T> = (args, extra) =>
+    traced(
+      async (span) => {
+        span.log({
+          input: args,
+          metadata: {
+            tool_name: toolDef.name,
+            tool_description: toolDef.description,
+          },
+        });
+
+        const result = await originalHandler(args, extra);
+
+        span.log({
+          output: result.content,
```

Comment:
looks like traced already collects the output. Should we just use the whole `result`? (trying to minimize massaging). That avoids the `is_error` in the metadata, for example.